### PR TITLE
implement newline lint

### PR DESCRIPTION
**Pull request by vapier**
Closes issue #682

### CODE REVIEW

File: pylint/checkers/format.py [hunk 1]
```diff
@@ -772,6 +776,8 @@ def process_tokens(self, tokens):
                 if len(indents) > 1:
                     del indents[-1]
             elif tok_type == tokenize.NL:
+                if line.strip('\r\n') is '':
```
Comment (The-Compiler) [inline pylint/checkers/format.py]: Please use `==`, not `is` to compare strings

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/.hg/
```
Comment (PCManticore) [inline .gitignore]: This needs to be separated from this PR.

**Comment by PCManticore** [comment]
Do you mind adding a ChangeLog entry and adding yourself to CONTRIBUTORS file?

**Comment by vapier** [comment]
updated the PR as requested

**Comment by PCManticore** [comment]
Thanks. I'll try to merge it in the following days, I want to setup Travis before this.

**Comment by PCManticore** [comment]
Merged locally with https://github.com/PyCQA/pylint/commit/f63bacb373bd98e4cbf01ed64d8a71b6df6a19ea

**Pull request closed without merging**